File: LivrusVirtus/wxWindows/wx.NET/Delphi.NET/fontes/ConAutores.pas
unit ConAutores;

interface

uses
  System.Drawing, wx, Consultas;

type
  TfrmConAutores = class(TfrmConsultas)
  private
    procedure OnClose(sender: System.Object; e: wx.Event);
    procedure OnBtnFechar(sender: System.Object; e: wx.Event);
    procedure OnBtnLimpar(sender: System.Object; e: wx.Event);
    procedure OnBtnPesquisar(sender: System.Object; e: wx.Event);
    procedure OnGridSelectCell(sender: System.Object; e: wx.Event);
    procedure OnGridLabelLeftClick(sender: System.Object; e: wx.Event);
  public
    constructor Create(parent: wx.Window);
  end;

var
  frmConAutores: TfrmConAutores;

  [assembly: RuntimeRequiredAttribute(TypeOf(TfrmConAutores))]

implementation

uses Resource, ConsultasSQL;

constructor TfrmConAutores.Create(parent: wx.Window);
begin
  inherited Create(parent, 'Autor:');

  Title := Res.STR_CONAUTOR;
  //labDescricao.Label := 'Autor:';
  gridConsulta.SetColLabelValue(1, 'Autor');

  EVT_BUTTON(ID_FRMCONSULTAS_BTNPESQUISAR, OnBtnPesquisar );
  EVT_BUTTON(ID_FRMCONSULTAS_BTNLIMPAR, OnBtnLimpar );
  EVT_BUTTON(ID_FRMCONSULTAS_BTNFECHAR, OnBtnFechar );
  EVT_CLOSE( OnClose );
  EVT_GRID_SELECT_CELL( OnGridSelectCell );
  EVT_GRID_LABEL_LEFT_CLICK( OnGridLabelLeftClick );
end;

procedure TfrmConAutores.OnClose(sender: System.Object; e: wx.Event);
begin
  Fecha(Res.MNU_CON_AUTORES);
	e.Skip();
end;

procedure TfrmConAutores.OnBtnFechar(sender: System.Object; e: wx.Event);
begin
  Fecha(0);
  e.Skip();
end;

procedure TfrmConAutores.OnBtnLimpar(sender: System.Object; e: wx.Event);
begin
  LimparDados();
  e.Skip();
end;

procedure TfrmConAutores.OnBtnPesquisar(sender: System.Object; e: wx.Event);
begin
  PesquisarDados(ConsSQL.Autor('S','', edtDescricao.Value),
    Res.STR_AUTNENC, 2);
  e.Skip();
end;

procedure TfrmConAutores.OnGridSelectCell(sender: System.Object; e: wx.Event);
var evt: wx.GridEvent;
begin
  evt := wx.GridEvent(e);
  setLabRegistros(evt.Row);
  e.Skip();
end;

procedure TfrmConAutores.OnGridLabelLeftClick(sender: System.Object;
  e: wx.Event);
var evt: wx.GridEvent;
begin
  evt := wx.GridEvent(e);
  setLabRegistros(evt.Row);
  e.Skip();
end;

end.

File: LivrusVirtus/VCPP6/wxWin/LibPostgreSQL/ConAutores.cpp
#include "resource.h"
#include "Consultas.h"
#include "Principal.h"
#include "ConsultasSQL.h"
#include "RotinasGlobais.h"
#include "ConAutores.h"

BEGIN_EVENT_TABLE(CfrmConAutores, CfrmConsultas)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNPESQUISAR, CfrmConAutores::OnBtnPesquisar)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNLIMPAR, CfrmConAutores::OnBtnLimpar)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNFECHAR, CfrmConAutores::OnBtnFechar)
  EVT_CLOSE(CfrmConAutores::OnClose)
  EVT_GRID_SELECT_CELL(CfrmConAutores::OnGridSelectCell)
  EVT_GRID_LABEL_LEFT_CLICK(CfrmConAutores::OnGridLabelLeftClick)
END_EVENT_TABLE()

CfrmConAutores::CfrmConAutores(wxWindow *parent)
       : CfrmConsultas(parent) {
	IniciaControles();
}

CfrmConAutores::~CfrmConAutores() { }

void CfrmConAutores::IniciaControles() {
 this->SetTitle(STR_CONAUTOR);
 this->labDescricao->SetLabel(wxT("Autor:  "));
 this->gridConsulta->SetColLabelValue(1, wxT("Autor"));
}

void CfrmConAutores::OnClose(wxCloseEvent& event) {
  Fecha(MNU_CON_AUTORES);
  event.Skip();
}

void CfrmConAutores::OnBtnFechar(wxCommandEvent& event) {  
  Fecha(0);
  event.Skip();
}

void CfrmConAutores::OnBtnLimpar(wxCommandEvent& event) {
  LimparDados();
  event.Skip();
}

void CfrmConAutores::OnBtnPesquisar(wxCommandEvent& event) {
  PesquisarDados(
    ConsSQL->Autor('S',wxT(""), edtDescricao->GetValue()), STR_AUTNENC, 2);
  event.Skip();
}

void CfrmConAutores::OnGridSelectCell(wxGridEvent& event) {
  setLabRegistros(event.GetRow());
  event.Skip();
}

void CfrmConAutores::OnGridLabelLeftClick(wxGridEvent& event){
  setLabRegistros(event.GetRow());
  event.Skip();
}

